### PR TITLE
Some docstring fixes/enhancements + Code cleanup

### PR DESCRIPTION
**Pull request by navidcy**

### CODE REVIEW

**Comment by navidcy** [comment]
@simone-silvestri the matrix poisson solver tests take quite a bit...

is this

https://github.com/CliMA/Oceananigans.jl/blob/9cd1a348aa3e720249e7569bfeaa7ce97d665211/test/test_matrix_poisson_solver.jl#L143

really needed to be 100x100? would 50x50 do the job?

**Comment by simone-silvestri** [comment]
Yeah we can change it also to 20X20

File: benchmark/benchmark_multithreading_single.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ using Benchmarks
 
 N = parse(Int, ARGS[1])
 grid = RectilinearGrid(size=(N, N, N), extent=(1, 1, 1))
-model = NonhydrostaticModel(architecture=CPU(), grid=grid)
+model = NonhydrostaticModel(architecture=CPU(); grid)
```
Comment (simone-silvestri) [inline benchmark/benchmark_multithreading_single.jl:10]: architecture is not an argument anymore, it's inferred from the grid
```suggestion
model = NonhydrostaticModel(; grid)
```

File: benchmark/benchmark_lagrangian_particle_tracking.jl [hunk 2]
```diff
@@ -20,7 +20,7 @@ function benchmark_particle_tracking(Arch, N_particles)
         particles = LagrangianParticles(x=x₀, y=y₀, z=z₀)
     end
 
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid, particles=particles)
+    model = NonhydrostaticModel(architecture=Arch(); grid, particles)
```
Comment (simone-silvestri) [inline benchmark/benchmark_lagrangian_particle_tracking.jl:23]:
```suggestion
    model = NonhydrostaticModel(; grid, particles)
```

File: benchmark/benchmark_equations_of_state.jl [hunk 3]
```diff
@@ -12,7 +12,7 @@ using Benchmarks
 function benchmark_equation_of_state(Arch, EOS)
     grid = RectilinearGrid(size=(192, 192, 192), extent=(1, 1, 1))
     buoyancy = SeawaterBuoyancy(equation_of_state=EOS())
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid, buoyancy=buoyancy)
+    model = NonhydrostaticModel(architecture=Arch(); grid, buoyancy)
```
Comment (simone-silvestri) [inline benchmark/benchmark_equations_of_state.jl:15]:
```suggestion
    model = NonhydrostaticModel(; grid, buoyancy)
```

File: benchmark/benchmark_advection_schemes.jl [hunk 4]
```diff
@@ -10,7 +10,7 @@ using Benchmarks
 
 function benchmark_advection_scheme(Arch, Scheme)
     grid = RectilinearGrid(size=(192, 192, 192), extent=(1, 1, 1))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid, advection=Scheme())
+    model = NonhydrostaticModel(architecture=Arch(); grid, advection=Scheme())
```
Comment (simone-silvestri) [inline benchmark/benchmark_advection_schemes.jl:13]:
```suggestion
    model = NonhydrostaticModel(; grid, advection=Scheme())
```

File: benchmark/benchmark_abstract_operations.jl [hunk 5]
```diff
@@ -26,7 +26,7 @@ for Arch in Archs
     N = Arch == CPU ? (32, 32, 32) : (256, 256, 256)
 
     grid = RectilinearGrid(FT, size=N, extent=(1, 1, 1))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid,
+    model = NonhydrostaticModel(architecture=Arch(); grid,
```
Comment (simone-silvestri) [inline benchmark/benchmark_abstract_operations.jl:29]:
```suggestion
    model = NonhydrostaticModel(; grid,
```

File: benchmark/benchmark_shallow_water_model.jl [hunk 6]
```diff
@@ -12,7 +12,7 @@ pyplot()
 
 function benchmark_shallow_water_model(Arch, FT, N)
     grid = RectilinearGrid(FT, size=(N, N), extent=(1, 1), topology=(Periodic, Periodic, Flat), halo=(3, 3))
-    model = ShallowWaterModel(architecture=Arch(), grid=grid, gravitational_acceleration=1.0)
+    model = ShallowWaterModel(architecture=Arch(); grid, gravitational_acceleration=1.0)
```
Comment (simone-silvestri) [inline benchmark/benchmark_shallow_water_model.jl:15]:
```suggestion
    model = ShallowWaterModel(; grid, gravitational_acceleration=1.0)
```

File: benchmark/benchmark_time_steppers.jl [hunk 7]
```diff
@@ -7,9 +7,9 @@ using Benchmarks
 
 # Benchmark function
 
-function benchmark_time_stepper(Arch, N, TimeStepper)
+function benchmark_time_stepper(Arch, N, timestepper)
     grid = RectilinearGrid(size=(N, N, N), extent=(1, 1, 1))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid, timestepper=TimeStepper)
+    model = NonhydrostaticModel(architecture=Arch(); grid, timestepper)
```
Comment (simone-silvestri) [inline benchmark/benchmark_time_steppers.jl:12]:
```suggestion
    model = NonhydrostaticModel(; grid, timestepper)
```

File: benchmark/benchmark_topologies.jl [hunk 8]
```diff
@@ -9,7 +9,7 @@ using Benchmarks
 
 function benchmark_topology(Arch, N, topo)
     grid = RectilinearGrid(topology=topo, size=(N, N, N), extent=(1, 1, 1))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid)
+    model = NonhydrostaticModel(architecture=Arch(); grid)
```
Comment (simone-silvestri) [inline benchmark/benchmark_topologies.jl:12]:
```suggestion
    model = NonhydrostaticModel(; grid)
```

File: benchmark/benchmark_tracers.jl [hunk 9]
```diff
@@ -31,7 +31,8 @@ end
 function benchmark_tracers(Arch, N, n_tracers)
     n_active, n_passive = n_tracers
     grid = RectilinearGrid(size=(N, N, N), extent=(1, 1, 1))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid, buoyancy=buoyancy(n_active),
+    model = NonhydrostaticModel(architecture=Arch(); grid,
```
Comment (simone-silvestri) [inline benchmark/benchmark_tracers.jl:34]:
```suggestion
    model = NonhydrostaticModel(; grid,
```

File: benchmark/benchmark_turbulence_closures.jl [hunk 10]
```diff
@@ -10,7 +10,7 @@ using Benchmarks
 
 function benchmark_closure(Arch, Closure)
     grid = RectilinearGrid(size=(128, 128, 128), extent=(1, 1, 1))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid, closure=Closure())
+    model = NonhydrostaticModel(architecture=Arch(); grid, closure=Closure())
```
Comment (simone-silvestri) [inline benchmark/benchmark_turbulence_closures.jl:13]:
```suggestion
    model = NonhydrostaticModel(; grid, closure=Closure())
```

File: benchmark/benchmark_vertically_stretched_nonhydrostatic_model.jl [hunk 11]
```diff
@@ -9,7 +9,7 @@ using Benchmarks
 
 function benchmark_vertically_stretched_nonhydrostatic_model(Arch, FT, N)
     grid = RectilinearGrid(architecture=Arch(), size=(N, N, N), x=(0, 1), y=(0, 1), z=collect(0:N))
-    model = NonhydrostaticModel(architecture=Arch(), grid=grid)
+    model = NonhydrostaticModel(architecture=Arch(); grid)
```
Comment (simone-silvestri) [inline benchmark/benchmark_vertically_stretched_nonhydrostatic_model.jl:12]:
```suggestion
    model = NonhydrostaticModel(; grid)
```

File: benchmark/distributed_nonhydrostatic_model_threaded.jl [hunk 12]
```diff
@@ -21,7 +21,7 @@ T = Threads.nthreads()
 
 topo = (Periodic, Periodic, Periodic)
 grid = RectilinearGrid(topology=topo, size=(Nx, Ny, Nz), extent=(1, 1, 1))
-model = NonhydrostaticModel(architecture=CPU(), grid=grid)
+model = NonhydrostaticModel(architecture=CPU(); grid)
```
Comment (simone-silvestri) [inline benchmark/distributed_nonhydrostatic_model_threaded.jl:24]:
```suggestion
model = NonhydrostaticModel(; grid)
```

File: docs/src/model_setup/buoyancy_and_equation_of_state.md [hunk 13]
```diff
@@ -157,7 +157,7 @@ SeawaterBuoyancy{Float64}:
 ├── gravitational_acceleration: 1.3
 └── equation of state: LinearEquationOfState(thermal_expansion=0.000167, haline_contraction=0.00078)
 
-julia> model = NonhydrostaticModel(; grid, buoyancy=buoyancy, tracers=(:T, :S))
+julia> model = NonhydrostaticModel(; grid, buoyancy, tracers=(:T, :S))
```
Comment (tomchor) [inline docs/src/model_setup/buoyancy_and_equation_of_state.md:160]: Why is `buoyancy=buoyancy` bad? In general I feel like keyword arguments are more user-friendly (with keywords a user doesn't need to know the order of arguments).

File: test/regression_tests/thermal_bubble_regression_test.jl [hunk 14]
```diff
@@ -11,7 +11,7 @@ function run_thermal_bubble_regression_test(arch, grid_type)
     end
 
     closure = ScalarDiffusivity(ν=4e-2, κ=4e-2)
-    model = NonhydrostaticModel(grid=grid, closure=closure, coriolis=FPlane(f=1e-4),
+    model = NonhydrostaticModel(; grid, closure, coriolis=FPlane(f=1e-4),
```
Comment (tomchor) [inline test/regression_tests/thermal_bubble_regression_test.jl:14]: Same comment as before. In my opinion `grid=grid, closure=closure` is more user friendly.

Comment (navidcy) [reply test/regression_tests/thermal_bubble_regression_test.jl:14]: hm... perhaps you have a point here

**Pull request closed without merging**